// swagger-ui-cli/dist/contracts.js
"use strict";
/**
 * swagger-ui-cli - Standalone Swagger UI server
 * Copyright (C) 2020  e.GO Digital GmbH, Aachen, Germany
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, version 3.
 *
 * This program is distributed in the hope that it will be useful, but
 * WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
 * General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CONTENT_TYPE_YAML = exports.CONTENT_TYPE_TOML = exports.CONTENT_TYPE_JSON = exports.CONTENT_TYPE_JAVASCRIPT = exports.NOT_SUPPORTED = exports.MIME_YAML = exports.MIME_TOML = exports.MIME_JSON = exports.MIME_JAVASCRIPT = exports.DEFAULT_CHARSET_HTTP = exports.DEFAULT_CHARSET = exports.ExitCode = void 0;
/**
 * List of known exit codes.
 */
var ExitCode;
(function (ExitCode) {
    /**
     * OK
     */
    ExitCode[ExitCode["OK"] = 0] = "OK";
    /**
     * Unhandled error
     */
    ExitCode[ExitCode["UncaughtError"] = 1] = "UncaughtError";
    /**
     * Showed help
     */
    ExitCode[ExitCode["ShowHelp"] = 2] = "ShowHelp";
    /**
     * No document source defined
     */
    ExitCode[ExitCode["NoDocumentDefined"] = 3] = "NoDocumentDefined";
    /**
     * Invalid port value.
     */
    ExitCode[ExitCode["InvalidPortValue"] = 4] = "InvalidPortValue";
    /**
     * Invalid document format
     */
    ExitCode[ExitCode["InvalidDocumentFormat"] = 5] = "InvalidDocumentFormat";
    /**
     * Invalid document type
     */
    ExitCode[ExitCode["InvalidDocumentType"] = 6] = "InvalidDocumentType";
    /**
     * Document source is no file
     */
    ExitCode[ExitCode["NoFile"] = 7] = "NoFile";
})(ExitCode = exports.ExitCode || (exports.ExitCode = {}));
/**
 * Defualt charset
 */
exports.DEFAULT_CHARSET = 'utf8';
/**
 * Default charset (HTTP header)
 */
exports.DEFAULT_CHARSET_HTTP = 'utf-8';
/**
 * MIME type JavaScript
 */
exports.MIME_JAVASCRIPT = 'text/javascript';
/**
 * MIME type JSON
 */
exports.MIME_JSON = 'application/json';
/**
 * MIME type TOML
 */
exports.MIME_TOML = 'application/toml';
/**
 * MIME type YAML
 */
exports.MIME_YAML = 'application/x-yaml';
/**
 * A value, that indicates, if something is NOT supported.
 */
exports.NOT_SUPPORTED = Symbol('NOT_SUPPORTED');
/**
 * Content-Type value for JavaScript files.
 */
exports.CONTENT_TYPE_JAVASCRIPT = `text/javascript; charset=${exports.DEFAULT_CHARSET_HTTP}`;
/**
 * Content-Type value for JSON files.
 */
exports.CONTENT_TYPE_JSON = `${exports.MIME_JSON}; charset=${exports.DEFAULT_CHARSET_HTTP}`;
/**
 * Content-Type value for TOML files.
 */
exports.CONTENT_TYPE_TOML = `${exports.MIME_TOML}; charset=${exports.DEFAULT_CHARSET_HTTP}`;
/**
 * Content-Type value for YAML files.
 */
exports.CONTENT_TYPE_YAML = `${exports.MIME_YAML}; charset=${exports.DEFAULT_CHARSET_HTTP}`;
